autopush_rs/src/client.rs

```diff
@@ -712,7 +712,7 @@ where
 
     pub fn shutdown(&mut self) {
         // If we made it past hello, do more cleanup
-        let webpush = match self.webpush.take() {
+        let mut webpush = match self.webpush.take() {
```

Comment:
one more nitpick: we don't have to clone stats below any longer now that webpush is mut

Reply:
The webpush.rx and webush.unacked* calls require mutable borrows.

Reply:
Oh, nevermind, I see what you mean.